KeyBasedBatchBuilder not work in perf code

#### Expected behavior

perf code will works well when batch builder set to KeyBasedBatchBuilder

#### Actual behavior

panic: multi batches container not support Flush(), please use FlushBatches() instead

#### Steps to reproduce

Set BatcherBuilderType to pulsar.KeyBasedBatchBuilder in [perf code](https://github.com/apache/pulsar-client-go/blob/master/perf/perf-producer.go#L88), run the perf with 

```shell
$ perf -r 1000 my-topic
```

#### System configuration
**Pulsar version**: any

--- a/pulsar/producer_partition.go
+++ b/pulsar/producer_partition.go
@@ -397,7 +397,11 @@ func (p *partitionProducer) internalSend(request *sendRequest) {
 		msg.ReplicationClusters, deliverAt)
 	if !added {
 		// The current batch is full.. flush it and retry
-		p.internalFlushCurrentBatch()
+		if p.batchBuilder.IsMultiBatches() {
+			p.internalFlushCurrentBatches()
+		} else {
+			p.internalFlushCurrentBatch()
+		}
 
 		// after flushing try again to add the current payload
 		if ok := p.batchBuilder.Add(smm, p.sequenceIDGenerator, payload, request,

